### ISSUE TITLE
angular 8 compatibility

### ISSUE THREAD

**Issue by joniham** [xtreamsrl/ngx-validation-errors#10]
When installing on ng8 project npm complains about angular @angular/core not being installed. However in ng8 the libraries exist with greater version numbers.

After that seems not to be working on ng8 project (no validation errors are shown, no errors neither). Also seems to be incompatible with angular material components

With: 
```
<mat-form-field formFieldContainer>
    <input matInput...>
</mat-form-field>
```
I got error: 
`ERROR Error: mat-form-field must contain a MatFormFieldControl.
    at getMatFormFieldMissingControlError (VM54712 vendor.js:111728)
    at MatFormField._validateControlChild (VM54712 vendor.js:112310)
    at MatFormField.ngAfterContentChecked (VM54712 vendor.js:112114)
    at callProviderLifecycles (VM54712 vendor.js:87970)
    at callElementProvidersLifecycles (VM54712 vendor.js:87939)
    at callLifecycleHooksChildrenFirst (VM54712 vendor.js:87921)
    at checkAndUpdateView (VM54712 vendor.js:98799)
    at callViewAction (VM54712 vendor.js:99160)
    at execComponentViewsAction (VM54712 vendor.js:99088)
    at checkAndUpdateView (VM54712 vendor.js:98801)`

Great idea for a easy global validation! Would be great to see ng8 and material compatibility fixed.

**Comment by rams23** [comment]
Hi @joniham I'll come back to you with a fix as soon as possible. Thanks for the feedback

**Comment by rams23** [comment]
About angular Material, it is an interesting case: the formFieldContainer is a component, so it is not applicable to mat-form-field because it is a component itself, but the automatic error addition is based on content projection and ViewContainerRef in formFieldContainer. So I have to find a way to access the mat-form-filed content and inject the mat-error, I actually don't know if it is possible.

**Comment by rams23** [comment]
@joniham I released a major version of the library v1.0.0 migrating to angular 8 and adding a structural directive to be compatible with angular material and similar libraries.  Here the usage (more info in the readme).

```angular2html
    <mat-form-field *ngxValidationErrors="heroForm.get('name'); errors as errors">
      <input matInput formControlName="name" placeholder="name"/>
      <mat-error *ngIf="errors">{{errors}}</mat-error>
    </mat-form-field>
```

You can find a complete form also in the material section of the example app

**Comment by joniham** [comment]
@rams23 Thank you so much for your effort!

Again, great little lib!
